cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/query/IndexType.java

```diff
@@ -0,0 +1,53 @@
+/*
+ *  Copyright (c) 2016 IBM Corp. All rights reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ *  except in compliance with the License. You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software distributed under the
+ *   License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ *  either express or implied. See the License for the specific language governing permissions
+ *  and limitations under the License.
+ */
+
+package com.cloudant.sync.query;
+
+/**
+ * Denotes the type for an Query Index.
+ */
+public enum IndexType {
+
+    /**
+     * JSON Index
+     */
+    JSON,
```

Comment:
How about calling these "json" and "text" - then their enum names would match the strings and you don't have to over-ride `toString` or provide `enumValue` (which appears to be doing a similar thing to the built-in `Enum.valueOf`)

The only downside is that it may break coding standards but in situations like these where we have to maintain compatibility with some existing strings I think it's worth it.